connectors/protocol/connectors.py

```diff
@@ -172,6 +173,18 @@ async def connector_put(
             index_name=index_name,
         )
 
+    async def connector_exists(self, connector_id):
+        try:
+            doc = await self.fetch_by_id(connector_id)
+            return doc is not None
```

Comment:
It's probably fine now, but is there a way to do this with as little traffic as possible? We can have connector records with cursors that'll weigh up to 5MB, personally I expect a method like "exists" to be veeeery lightweight

Reply:
Agreed, it should be lightweight, there is not much we can do in this PR as we would need to adapt the API to allow to query only for selected fields.

I filed an issue: https://github.com/elastic/connectors/issues/2937